### PR TITLE
[#5473] Aggregation meta file download fix

### PR DESCRIPTION
**Pull request by pkdash**
### Pull Request Checklist: 
- [x] Positive Test Case Written by Dev

- [x] Automated Testing

- [x] Sufficient User and Developer Documentation

- [x] Passing Jenkins Build

- [x] Peer Code review and approval

### Positive Test Case
1. Refer to issue #5473 to test.

### CODE REVIEW

File: hs_file_types/models/base.py [hunk 1]
```diff
@@ -25,10 +25,7 @@
     set_dirty_bag_flag, add_file_to_resource, resource_modified, get_file_from_irods
 from hs_core.models import ResourceFile, AbstractMetaDataElement, Coverage
 from hs_core.signals import post_remove_file_aggregation
-
-RESMAP_FILE_ENDSWITH = "_resmap.xml"
-METADATA_FILE_ENDSWITH = "_meta.xml"
-SCHEMA_JSON_FILE_ENDSWITH = "_schema.json"
```
Comment (devincowan) [inline hs_file_types/models/base.py:31]: might want to update the `is_map_file()` here:
https://github.com/hydroshare/hydroshare/blob/5473-aggregation-meta-file-download-fix/hs_file_types/utils.py#L314-L315

File: hs_composite_resource/models.py [hunk 2]
```diff
@@ -333,6 +345,38 @@ def get_aggregation_by_name(self, name, aggregations=None):
...
+        for res_file in ResourceFile.list_folder(self, folder=folder, sub_folders=False):
+            if res_file.has_logical_file:
+                file_name_to_match = f"{base_file_name_to_match}{res_file.extension}"
+                if res_file.file_name == file_name_to_match:
+                    return res_file.logical_file
+        # check for folder aggregation
+        if folder and folder == base_file_name_to_match:
+            folder_aggregation = self.get_folder_aggregation_object(folder)
+            if folder_aggregation is not None:
+                return folder_aggregation
```
Comment (devincowan) [inline hs_composite_resource/models.py:375]: in one case we return a logical file, in the other case we return a folder_aggregation. Don't we expect to always return an aggregation?

**Review by devincowan — APPROVED** [review]

File: hs_composite_resource/models.py (hunk 2 above)
Comment (pkdash) [reply hs_composite_resource/models.py:375]: logical_file is same as aggregation. In the database it is logical_file.

File: hs_file_types/models/base.py (hunk 1 above)
Comment (pkdash) [reply hs_file_types/models/base.py:31]: I assume use the new enum types for aggregation meta file extensions to replace the '_resmap.xml'? Unless we also replace the 'resourcemap.xml' using another enum type, I think it is better to not change.

Comment (devincowan) [reply hs_file_types/models/base.py:31]: sounds good!

File: hs_composite_resource/models.py (hunk 2 above)
Comment (devincowan) [reply hs_composite_resource/models.py:375]: I see it's either a folder aggregation or a logicalFile aggregation. Thanks!

**Pull request merged**